vision/pipeline.py

```diff
@@ -88,54 +95,116 @@ def run(self, prev_state):
+                except:
+                    flags.obstacle_tracker = False
 
         elif state == "text_detection":  # approaching mast
-            bboxes = self.text_detector.detect_russian_word(color_image, depth_image)
+            flags = TextDetectionFlags()
+
+            try:
+                bboxes = self.text_detector.detect_russian_word(color_image, depth_image)
+            except:
+                flags.detect_russian_word = False
 
         elif state == "module_detection":  # locating module
-            self.module_location.set_img(color_image, depth_image)
-
-            # only do more calculation if module is in the image
-            if self.module_location.is_in_frame():
-                center = self.module_location.get_center()  # center of module in image
-                depth = get_module_depth(
-                    depth_image, center
-                )  # depth of center of module
-
-                region = region_of_interest(
-                    depth_image, depth, center
-                )  # depth image sliced on underestimate bounds
-                orientation = get_module_orientation(
-                    region, center
-                )  # x and y tilt of module
-
-                bounds = get_module_bounds(
-                    color_image, center, depth
-                )  # overestimate of bounds
-                roll = get_module_roll(
-                    color_image[
-                        bounds[0][1] : bounds[3][1], bounds[0][0] : bounds[2][0], :
-                    ]
-                )  # roll of module
-
-                # construct boundingbox for the module
-                box = BoundingBox(bounds, ObjectType.MODULE)
-                box.module_depth = depth  # float
-                box.orientation = orientation + (roll,)  # x, y, z tilt
-                bboxes.append(box)
+            flags = ModuleDetectionFlags()
+            try:
+                self.module_location.setImg(color_image, depth_image)
+            except:
+                flags.set_img = False
+
+            if flags.set_img:
+                try:
+                    inFrame = self.module_location.is_in_frame()
+                except:
+                    inFrame = False
+                    flags.is_in_frame = False
+
+                # only do more calculation if module is in the image
+                if inFrame:
+                    # default values for bounding box construction
+                    center = (0, 0)
+                    roll = 0
+                    depth = 0
+                    region = np.empty(1)
+                    orientation = (0, 0)
+                    bounds = np.empty(1)
+
+                    try:
+                        center = self.module_location.get_center()  # center of module in image
+                    except:
+                        flags.get_center = False
+
+                    try:
+                        roll = get_module_roll(
```

Comment:
Not sure why the module roll function got moved up here. Should be run after get_module_bounds since it depends on that. Right now it would be run on the default value of bounds.